src/Microsoft.AspNetCore.SignalR.Common/Protocol/HandshakeProtocol.cs

```diff
@@ -235,6 +236,13 @@ public static bool TryParseRequestMessage(ref ReadOnlySequence<byte> buffer, out
                 Utf8BufferTextReader.Return(textReader);
             }
 
+            // For error messages, we want to print the payload as text
+            string GetPayloadAsString()
+            {
+                // REVIEW: Should we show hex for binary charaters?
```

Comment:
I mean, you're only doing this in the HandshakeProtocol which is always JSON, so .... no?

Reply:
If you sent some BS (like a message pack invocation before the handshake), it would be useful. But I don't think it matters much.